### ISSUE TITLE
Rationalize Perseus idioms

### ISSUE THREAD

**Issue by arctic-hen7** [arctic-hen7/perseus#32]
**Is your feature request related to a problem? Please describe.**
If you take three seconds to go through the Perseus idiom of defining functions that return closures within `Rc`s, you'll probably stop liking it pretty quickly, because it's grotesque. Besides, we can trivially just have those closures as functions themselves that are then placed in `Rc`s (which is also an idiom, but only for some things...).

**Describe the solution you'd like**
In short, these idioms need to start making sense! A lot of them are left over from the old architecture of v0.1.0 and even before that, and they should be updated in the documentation and examples to reflect better practice (which is more concise and logical).

**Describe alternatives you've considered**

**Additional context**
This is primarily a documentation change, but it's a very big one, and it'll be released in v0.3.0.

**Comment by arctic-hen7** [comment]
I'll work on this with the changes in #31.
